create Header component

DeeDowns opened the pull request:
## Description
<!-- What does this code change? Why did I choose this approach? Did I learn anything worth sharing? Reminder: This will be a publicly facing representation of your work (READ: help you land that sweet dev gig). -->
Adds header and top border with styling to outline the main part of the app

## Related Issue
<!-- If you write "closes" followed by the Github issue number, it will automatically close the issue for you when the PR merges -->
closes[#62]

## Acceptance Criteria
<!-- Include AC from the Github issue -->

- [x] There is a header element to separate the title and logo from main part of the app

- [x] There is a border to separate navigation elements from main part of the app

- [x] The header background color and border color are be the same

- [x] The `List` and `Add Item` pages have the same styles applied to them

## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates


### Before
<!-- If UI feature, take provide screenshots -->
![header-nav](https://user-images.githubusercontent.com/47228065/109234067-fdb27f00-7798-11eb-95c6-f7fe5a9bd626.gif)


### After
<!-- If UI feature, take provide screenshots -->
![header-nav-styled](https://user-images.githubusercontent.com/47228065/109245193-06fa1680-77ae-11eb-8fe2-05ba37bf4769.gif)

## Testing Steps / QA Criteria
<!-- Provide steps the other cohort members and mentors need to follow to properly test your additions. -->

From your terminal, pull down this branch with `git pull origin DD-header-nav-outline` and check that branch out with `git checkout DD-header-nav-outline`
Then type `npm start` in your terminal to launch the app.

OR
Click on the details button below to open the Netlify preview
<img width="660" alt="Screen Shot 2021-02-23 at 9 08 15 PM" src="https://user-images.githubusercontent.com/47228065/109234195-3d796680-7799-11eb-9b23-1339bf655eb9.png">

- Any token will do (`calf bebop styx`)

- Navigate to the Add Item or List page to view the header and styling above the navigation elements

DeeDowns commented on src/components/List.css:
```diff
@@ -1,3 +1,11 @@
+.list-container {
```
I had to widen the top level div for the List and Add Item page so the header would span the entire width of the page

LINDGRENBA replied:
Another way to possibly tackle that would be to create a Layout that wraps around App and put the header and footer in the layout. But I think if the header issue in the mobile view is resolved based on what you and Lauren talked about below then that's unnecessary.

DeeDowns commented on src/components/List.js:
```diff
@@ -136,8 +137,8 @@ const List = ({ token }) => {
-    <div>
-      <h1>List</h1>
+    <div className="list-container">
+      <Header />
```
I added the Header component to the List and Add Item components and not the App component, because the Welcome page will have its own styling and won't include the header or navigation elements.

DeeDowns commented on src/styles/Header.css:
```diff
@@ -0,0 +1,16 @@
+  font-size: 2.4rem;
+}
+
+header .homeIcon {
```
Let me know if you think the size of the logo and/or title font size should change

laurenmbeatty commented on src/components/AddItem.js:
```diff
@@ -105,7 +108,7 @@ const AddItem = ({ token }) => {
         <button type="submit">
           <span>
-            Add Item<i class="fas fa-cart-plus"></i>
+            Add Item<i className="fas fa-cart-plus"></i>
```
Thank you!

laurenmbeatty commented on src/components/Header.js:
```diff
@@ -0,0 +1,18 @@
+      <img
+        src={homeIcon}
+        className="homeIcon"
+        alt="colorful grocery bag logo and home link"
```
excellent

laurenmbeatty commented:
@DeeDowns Use the dev tools to give a mobile view. This is what I see on the List page:

![image](https://user-images.githubusercontent.com/10378331/109247916-60f8dd00-77aa-11eb-803f-debc3952f200.png)

Just need to toggle here:
![image](https://user-images.githubusercontent.com/10378331/109247967-82f25f80-77aa-11eb-989d-73445c725701.png)

laurenmbeatty commented:
Having said that, if I navigate to the Add Item page, and then back to the list, then it looks OK. You all might want to consider putting the list in a scrolling container, so that the navigation is always at the bottom of the screen.

Oh, I see why. Something in the list items has a width of 385px...could just make the width of the ul be like 90% or something, and then get rid of that specific width.

laurenmbeatty commented:
Then it would look like this:
![image](https://user-images.githubusercontent.com/10378331/109248796-1aa47d80-77ac-11eb-90f0-1d02d9025e88.png)

laurenmbeatty reviewed (approved):
Looks cute!  Probably want to create a separate issue for the list items.

DeeDowns commented:
> Having said that, if I navigate to the Add Item page, and then back to the list, then it looks OK. You all might want to consider putting the list in a scrolling container, so that the navigation is always at the bottom of the screen.
> 
> Oh, I see why. Something in the list items has a width of 385px...could just make the width of the ul be like 90% or something, and then get rid of that specific width. Would also need to get rid of the padding on the ul...unordered lists have 15px of left padding built in.
> 
> You could just create a separate issue for that, since it's not part of this ticket. 👍

Oh ok so I changed the `.flex-direction` width to 100% and added the `ul` width to be 25% and that fixed the white showing when you resize to mobile view. I'm not going to push it, but will make a note for when we get to media queries and responsive styling.
I think this was from another issue, I normally use % instead of px for width. Is it better to use one over the other?

MikeBLambert reviewed (approved):
Header looks great! 
You may want to consider rendering the Header component making the \<Header /\> component part of the \<App /\> and only rendering it if there is a token. 
That, or you could make something like a \<Page /\> component, which renders a div that has the css that's duplicated for both the .list-container and the .add-item-container, with the \<Header /\> 
It would look something like 
const Page = ({children}) => 
\<div className="page"\>
\<Header /\>
    {children}
    \</div\>

Then, rather than a \<div /\> container in \<List /\> and \<AddItem /\> you use your \<Page /\> component and...voila! 
Just an idea. 
Again, what you've got here looks great!

laurenmbeatty commented:
> Oh ok so I changed the `.flex-direction` width to 100% and added the `ul` width to be 25% and that fixed the white showing when you resize to mobile view. I'm not going to push it, but will make a note for when we get to media queries and responsive styling.
> I think this was from another issue, I normally use % instead of px for width. Is it better to use one over the other?

Yeah, flexible whenever possible, for sure. Sometimes we need fixed size, but i usually try to go for flexible.

LINDGRENBA reviewed (approved):
It looks really good Dee! Everything is consistent and I like that you figured out how we can resolve the header issue in mobile view, but made the decision to leave that for the issue related to the mobile view!
